dips/dip-158.md

```diff
@@ -0,0 +1,805 @@
+* `risk_checks_failed`- This abort code should be used when the risk checks performed based on the payer information failed
+* `payment_type_mismatch` - This abort code should be used when there is  a mismatch between the command type and the payment type. For example, if the sender initiated InitChargeCommand for a payment of type auth
+* `rejected` - This abort code should be used for aborting a payment not covered by other values of the enum
+
+## AddressObject
+Represents a physical address
+| Field | Type | Required? | Description |
+|-|-|-|-|
+| city | str | Y | The city, district, suburb, town, or village |
+| country | str | Y | Two-letter country code (https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) |
+| line1 | str | Y | Address line 1 |
+| line2 | str | N | Address line 2 - apartment, unit, etc.|
+| postal_code| str | Y | ZIP or postal code |
+| state | str | N | State, county, province, region.|
+```
+{
+    "city": "San Francisco",
+    "country": "US",
+    "line1": "1260 Market Street",
+    "line2": "Suite 450",
+    "postal_code": "94103",
+    "state": "CA",
+}
+```
+## NationalIdObject
+Represents a national ID.
+
+| Field | Type | Required? | Description |
+|-|-|-|-|
+| id_value | str | Y | Indicates the national ID value - for example, a social security number |
+| country | str | N | Two-letter country code (https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) |
+| type | str | N | Indicates the type of the ID |
+
+```
+{
+    "id_value": "123-45-6789",
+    "country": "US",
+    "type": "SSN",
+}
+```
+
+---
+# Appendix A - Prerequisite: Sharing Common Payment Identifier and Address
+
+Note that this is a prerequisite for the off-chain negotiations and not a part of it.
+
+## Acquire Payment Identifier and Receiver Address Using Redirect URL
+
+This URL contains the payment reference id (i.e. the common payment identifier) and the wallet's website address.
+
+The redirect URL is likely to be used in the **Web Redirect** flow which means that the customer will be redirected from the merchant's checkout page to the wallet's website to review and approve the payment. 
+
+The table below specifies the fields that should be encoded into a series of URL parameters appended to the query string.
+
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|vasp_address|str|Y|Address of receiving VASP. The address is encoded using bech32. For Diem addresses format, refer to the "account identifiers" section in [DIP-5](https://dip.diem.com/dip-5/#account-identifiers).|
+|reference_id|str|Y|A unique identifier of this payment. It should be a UUID according to RFC4122 with "-"'s included.|
+|redirect_url|str|N|URL used by the wallet to redirect the customer back to the merchant|
```

Comment:
I guess the intent here is that the merchant could bake whatever they want here?

Reply:
Yes.
At the end of the payment process, the wallet will have to redirect the customer back to the merchant to show a confirmation page (or failure page). The redirect_url is intended for that purpose, i.e. the customer will be redirected, the merchant will verify the payment status, and will display the appropriate page